### PR TITLE
[BEAM-1792] Changing Spark to use MetricFiltering

### PR DESCRIPTION
**Pull request by pabloem**
* [x] Make sure the PR title is formatted like: [BEAM-<Jira issue #>] Description of pull request
* [x] Make sure tests pass via mvn clean verify. (Even better, enable Travis-CI on your fork and ensure the whole test matrix passes).
* [x] Replace <Jira issue #> in the title with the actual Jira issue number, if there is one.
* [x] If this contribution is large, please file an Apache Individual Contributor License Agreement.

### CODE REVIEW

**Comment by pabloem** [comment]
r: @aviemzur 
hi Aviem. Can you take a look? It's a fairly simple change.

**Comment by aviemzur** [comment]
Run Spark RunnableOnService

File: sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricNameFilter.java [hunk 1]
```diff
@@ -54,7 +54,6 @@ public static MetricNameFilter named(String namespace, String name) {
   public static MetricNameFilter named(Class<?> namespace, String name) {
     checkNotNull(namespace, "Must specify a inNamespace");
     checkNotNull(name, "Must specify a name");
-    return new AutoValue_MetricNameFilter(namespace.getSimpleName(), name);
+    return new AutoValue_MetricNameFilter(namespace.getName(), name);
```
Comment (aviemzur) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricNameFilter.java]: This seems to be a positive change, but I'm not sure it is needed for this PR.

I'm assuming this change was done to match https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricName.java#L45-L45

There seems to be no test coverage for this. I'd add tests as well (Although again, probably in a separate PR).

Comment (pabloem) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricNameFilter.java]: That's fair. I've split this up into separate PRs.

**Comment by pabloem** [comment]
@aviemzur the build is fine now, it seems. If it looks fine to you, could you merge it? (If not, I can ask Ben, just let me know if the changes look okay ;)

**Comment by aviemzur** [comment]
It looks great!
Our build is still broken. We have a fix in https://github.com/apache/beam/pull/2314. After that is merged we can run the Spark tests again here and when they are green I'll merge.

**Comment by aviemzur** [comment]
Hi @pabloem 
#2314 has been merged, so we can now properly test your PR.
Can you please rebase on top of master and push to this branch?

**Comment by aviemzur** [comment]
Run Spark RunnableOnService

**Comment by aviemzur** [comment]
LGTM
